### PR TITLE
MSFT: 1251032 [PowerShellGallery] Nuget metadata extraction 

### PR DESCRIPTION
**Pull request by darek-msft**
This adds PowerShell metadata extraction from psd1 module manifest files.

### CODE REVIEW

File: src/NuGet.Services.Publish/PowerShellHashtableVisitor.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        public object VisitMergingRedirection(MergingRedirectionAst mergingRedirectionAst) { throw new UnexpectedElementException(); }
+        public object VisitExpandableStringExpression(ExpandableStringExpressionAst expandableStringExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitIndexExpression(IndexExpressionAst indexExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitAttributedExpression(AttributedExpressionAst attributedExpressionAst) { throw new UnexpectedElementException(); }
+        public object VisitBlockStatement(BlockStatementAst blockStatementAst) { throw new UnexpectedElementException(); }
+        public object VisitInvokeMemberExpression(InvokeMemberExpressionAst invokeMemberExpressionAst) { throw new UnexpectedElementException(); }
+
+        public object VisitStatementBlock(StatementBlockAst statementBlockAst)
+        {
+            return statementBlockAst?.Statements?.FirstOrDefault()?.Visit(this);
```
Comment (johnataylor) [inline src/NuGet.Services.Publish/PowerShellHashtableVisitor.cs]: I'm getting compile errors on this line.

Comment (darek-msft) [inline src/NuGet.Services.Publish/PowerShellHashtableVisitor.cs]: Are you using the newest compiler? Null propagation operator is a new C# 6 language feature.

Comment (johnataylor) [inline src/NuGet.Services.Publish/PowerShellHashtableVisitor.cs]: No unfortunately. I'm sure we will change soon. But for now can you back put this one.

**Comment by darek-msft** [comment]
I have replaced null propagation with manual null checks.

**Comment by darek-msft** [comment]
This is an updated pull request to accommodate PS renaming and recent Nuget changes.

**Pull request merged**